### ISSUE TITLE
Plugin Error Prevents Opening any Camunda BPMN File

### ISSUE THREAD

**Issue by blakeney** [GDNext-GmbH/polybpmn#13]
**Describe the bug**
When I attempt to open any Camunda BPMN file using PolyBPMN, I encounter a plugin exception in IntelliJ, and the file does not open.

**To Reproduce**
1. Open a project in IntelliJ that contains a Camunda BPMN file
2. Double-click the file in the Project window
3. See the error below

```
com.intellij.diagnostic.PluginException: Cannot create editor by provider com.gdnext.bpmn.intellij.plugin.polybpmn.b.a [Plugin: polybpmn-intellij-visualizer]
	at com.intellij.openapi.fileEditor.impl.EditorCompositeModelManager$fileEditorWithProviderFlow$editorsWithProviders$1$1$1.invokeSuspend(EditorCompositeModelManager.kt:140)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:608)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:873)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:763)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:750)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:118)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at com.gdnext.bpmn.intellij.plugin.core.c.a.c.invokeSuspend(Unknown Source)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:277)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:111)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$BuildersKt__BuildersKt(Builders.kt:84)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlockingWithParallelismCompensation(Builders.kt:60)
	at kotlinx.coroutines.BuildersKt.runBlockingWithParallelismCompensation(Unknown Source)
	at kotlinx.coroutines.internal.intellij.IntellijCoroutines.runBlockingWithParallelismCompensation(intellij.kt:41)
	at com.intellij.openapi.progress.CoroutinesKt$runBlockingCancellable$1.invoke(coroutines.kt:146)
	at com.intellij.openapi.progress.CoroutinesKt$runBlockingCancellable$1.invoke(coroutines.kt:139)
	at com.intellij.openapi.progress.ContextKt.prepareThreadContext(context.kt:85)
	at com.intellij.openapi.progress.CoroutinesKt.runBlockingCancellable(coroutines.kt:139)
	at com.intellij.openapi.progress.CoroutinesKt.runBlockingCancellable(coroutines.kt:122)
	at com.gdnext.bpmn.intellij.plugin.core.c.a.b$a.a(Unknown Source)
	at com.gdnext.bpmn.intellij.plugin.core.c.a.b.createEditorAsync(Unknown Source)
	at com.intellij.openapi.fileEditor.AsyncFileEditorProvider$createFileEditor$builder$1.invoke(AsyncFileEditorProvider.kt:37)
	at com.intellij.openapi.fileEditor.AsyncFileEditorProvider$createFileEditor$builder$1.invoke(AsyncFileEditorProvider.kt:37)
	at com.intellij.openapi.application.rw.InternalReadAction.insideReadAction(InternalReadAction.kt:104)
	at com.intellij.openapi.application.rw.InternalReadAction.access$insideReadAction(InternalReadAction.kt:15)
	at com.intellij.openapi.application.rw.InternalReadAction$tryReadCancellable$2.invoke(InternalReadAction.kt:94)
	at com.intellij.openapi.application.rw.InternalReadAction$tryReadCancellable$2.invoke(InternalReadAction.kt:93)
	at com.intellij.openapi.application.rw.CancellableReadActionKt.cancellableReadActionInternal$lambda$2$lambda$1$lambda$0(cancellableReadAction.kt:31)
	at com.intellij.openapi.application.impl.AnyThreadWriteThreadingSupport.tryRunReadAction(AnyThreadWriteThreadingSupport.kt:291)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:965)
	at com.intellij.openapi.application.rw.CancellableReadActionKt.cancellableReadActionInternal$lambda$2$lambda$1(cancellableReadAction.kt:29)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtilService.runActionAndCancelBeforeWrite(ProgressIndicatorUtilService.java:66)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runActionAndCancelBeforeWrite(ProgressIndicatorUtils.java:155)
	at com.intellij.openapi.application.rw.CancellableReadActionKt.cancellableReadActionInternal(cancellableReadAction.kt:27)
	at com.intellij.openapi.application.rw.InternalReadAction.tryReadCancellable(InternalReadAction.kt:93)
	at com.intellij.openapi.application.rw.InternalReadAction.tryReadAction(InternalReadAction.kt:77)
	at com.intellij.openapi.application.rw.InternalReadAction.readLoop(InternalReadAction.kt:64)
	at com.intellij.openapi.application.rw.InternalReadAction.access$readLoop(InternalReadAction.kt:15)
	at com.intellij.openapi.application.rw.InternalReadAction$runReadAction$4.invokeSuspend(InternalReadAction.kt:43)
	at com.intellij.openapi.application.rw.InternalReadAction$runReadAction$4.invoke(InternalReadAction.kt)
	at com.intellij.openapi.application.rw.InternalReadAction$runReadAction$4.invoke(InternalReadAction.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:62)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:163)
	at kotlinx.coroutines.BuildersKt.withContext(Unknown Source)
	at com.intellij.openapi.application.rw.InternalReadAction.runReadAction(InternalReadAction.kt:39)
	at com.intellij.openapi.application.rw.PlatformReadWriteActionSupport.executeReadAction(PlatformReadWriteActionSupport.kt:38)
	at com.intellij.openapi.application.ReadWriteActionSupport.executeReadAction$default(ReadWriteActionSupport.kt:15)
	at com.intellij.openapi.application.CoroutinesKt.constrainedReadAction(coroutines.kt:58)
	at com.intellij.openapi.application.CoroutinesKt.readAction(coroutines.kt:25)
	at com.intellij.openapi.fileEditor.AsyncFileEditorProvider.createFileEditor$suspendImpl(AsyncFileEditorProvider.kt:37)
	at com.intellij.openapi.fileEditor.AsyncFileEditorProvider.createFileEditor(AsyncFileEditorProvider.kt)
	at com.intellij.openapi.fileEditor.impl.EditorCompositeModelManager$fileEditorWithProviderFlow$editorsWithProviders$1$1$1.invokeSuspend(EditorCompositeModelManager.kt:120)
	... 6 more
Caused by: java.lang.IllegalStateException: Should not be called
	at com.intellij.openapi.fileEditor.AsyncFileEditorProvider.createEditorAsync(AsyncFileEditorProvider.kt:27)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	... 54 more

```

**BPMN definition**
The following simple BPMN diagram is sufficient to trigger the error, but it seems to happen with any Camunda BPMN diagram. I did not test with non-Camunda BPMN.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<bpmn:definitions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:bpmn="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:dc="http://www.omg.org/spec/DD/20100524/DC" xmlns:di="http://www.omg.org/spec/DD/20100524/DI" id="Definitions_1" targetNamespace="http://bpmn.io/schema/bpmn">
  <bpmn:process id="Process_1" isExecutable="false">
    <bpmn:startEvent id="StartEvent_1">
      <bpmn:outgoing>Flow_12u7su1</bpmn:outgoing>
    </bpmn:startEvent>
    <bpmn:task id="Activity_0j8h20a" name="Test">
      <bpmn:incoming>Flow_12u7su1</bpmn:incoming>
      <bpmn:outgoing>Flow_0d9ded0</bpmn:outgoing>
    </bpmn:task>
    <bpmn:sequenceFlow id="Flow_12u7su1" sourceRef="StartEvent_1" targetRef="Activity_0j8h20a" />
    <bpmn:endEvent id="Event_0qak5ln">
      <bpmn:incoming>Flow_0d9ded0</bpmn:incoming>
    </bpmn:endEvent>
    <bpmn:sequenceFlow id="Flow_0d9ded0" sourceRef="Activity_0j8h20a" targetRef="Event_0qak5ln" />
  </bpmn:process>
  <bpmndi:BPMNDiagram id="BPMNDiagram_1">
    <bpmndi:BPMNPlane id="BPMNPlane_1" bpmnElement="Process_1">
      <bpmndi:BPMNShape id="_BPMNShape_StartEvent_2" bpmnElement="StartEvent_1">
        <dc:Bounds x="173" y="102" width="36" height="36" />
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape id="Activity_0j8h20a_di" bpmnElement="Activity_0j8h20a">
        <dc:Bounds x="260" y="80" width="100" height="80" />
        <bpmndi:BPMNLabel />
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape id="Event_0qak5ln_di" bpmnElement="Event_0qak5ln">
        <dc:Bounds x="412" y="102" width="36" height="36" />
      </bpmndi:BPMNShape>
      <bpmndi:BPMNEdge id="Flow_12u7su1_di" bpmnElement="Flow_12u7su1">
        <di:waypoint x="209" y="120" />
        <di:waypoint x="260" y="120" />
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNEdge id="Flow_0d9ded0_di" bpmnElement="Flow_0d9ded0">
        <di:waypoint x="360" y="120" />
        <di:waypoint x="412" y="120" />
      </bpmndi:BPMNEdge>
    </bpmndi:BPMNPlane>
  </bpmndi:BPMNDiagram>
</bpmn:definitions>
```

**Expected behavior**
I expected the file to open as a diagram, allowing me to view/edit it.

**Screenshots**
See stacktrace in lieu of screenshots.

**IntelliJ version (please complete the following information):**
 - Edition: Ultimate
 - Version 2024.2.0.2
 
**Additional context**
 I do not know whether the bug might be platform-specific, but I am using Windows 11.

**Comment by valb3r** [comment]
@blakeney Thanks for reporting, this is the duplicate of https://github.com/GDNext-GmbH/polybpmn/issues/12 that is expected to be fixed with 1.3.1 that is scheduled for release

**Comment by blakeney** [comment]
Thanks @valb3r! I checked the list of open issues before I submitted this one, but I did not think to check the list of closed issues.
